Source/Core/Common/Subscribable.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+
+#include "Common/NonCopyable.h"
+
+template <typename... Args>
+class Subscribable : NonCopyable
+{
+public:
+  using SubscriptionID = uint32_t;
+
+  void Send(const Args&... args)
```

Comment:
This was actually correct, but I didn't specify the function type I wanted correctly. I've changed it to 

```c++
using CallbackType =   
 std::function<void(std::add_lvalue_reference_t<std::add_const_t<std::decay_t<Args>>>...)>;
```